### DISCUSSION TITLE
请问在pse版本的shuffle 证明是做什么用的？

### DISCUSSION THREAD

**Discussion by mdj33** [z2o-k7e/halo2-step-by-step#35] [Q&A]
Add a shuffle argument for some input expressions and table expressions.

**Comment by pochenai** [comment]
你指的是这个https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_proofs/examples/shuffle.rs？
Shuffle指的是一列的某些行，是另一列的某些行的重排序, 即A=Shuffle(B)；而 lookup 只是证明一列的某些行(A)属于另一列(B)，但并不需要强制要求A=B，只需要保证A∈B
